src/scheduler/queue_info.cpp

```diff
@@ -568,19 +532,16 @@ queue_info::queue_info(const char *qname): name(qname)
  */
 
 void
-free_queues(queue_info **qarr)
+free_queues(std::vector<queue_info *> &qarr)
```

Comment:
This function will delete all the queue_info *s, but will not actually clear the vector, is that ok? You'll have a vector of freed pointers. Maybe add a .clear()?